sdk/storage/Azure.Storage.Files.DataLake/src/RolePermissionsExtensions.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Text;
+using Azure.Storage.Files.DataLake.Models;
+
+namespace Azure.Storage.Files.DataLake
+{
+    /// <summary>
+    /// Extension methods for RolePermissions.
+    /// </summary>
+    public static class RolePermissionsExtensions
+    {
+        /// <summary>
+        /// Parses octal char to RolePermissions.
+        /// </summary>
+        public static RolePermissions ParseOctal(char octal)
+        {
+            RolePermissions rolePermissions = RolePermissions.All;
+
+            int value = Convert.ToInt32(octal);
+
+            if (value < 0 || value > 7)
+            {
+                Errors.InvalidArgument(nameof(octal));
+            }
+
+            if (value / 4 > 0)
+            {
+                rolePermissions |= RolePermissions.Read;
+            }
+            value %= 4;
+
+            if (value / 2 > 0)
+            {
+                rolePermissions |= RolePermissions.Write;
+            }
+            value %= 2;
+
+            if (value > 0)
+            {
+                rolePermissions |= RolePermissions.Execute;
+            }
```

Comment:
Probably best to do bit masks instead to check

Reply:
done.